src/widgets/SimpleTable.js

```diff
@@ -20,90 +21,100 @@ import { HeaderCell, HeaderRow } from './DataTable/index';
-
-      return (
-        <SimpleRow
-          rowData={item}
-          rowIndex={index}
-          rowKey={id}
-          style={rowStyle}
-          renderCells={renderCells}
-          onPress={selectRow}
-          isSelected={isSelected}
-        />
-      );
-    },
-    [extraData]
-  );
-
-  const renderHeaderCells = useCallback(
-    () =>
-      columns.map(({ title, width, alignText }, index) => (
-        <HeaderCell
-          title={title}
-          containerStyle={headerCells[alignText]}
+      return columns.map(({ key, width, alignText }, index) => (
+        <Cell
+          value={rowData[key]}
+          viewStyle={cellContainer[alignText]}
           textStyle={cellText[alignText]}
           isLastCell={index === columns.length - 1}
           width={width}
+          key={`${id}${key}`}
         />
-      )),
-    [columns]
-  );
+      ));
+    }, []);
 
-  const renderHeaders = useCallback(
-    () => <HeaderRow style={headerRow} columns={columns} renderCells={renderHeaderCells} />,
-    [columns]
-  );
+    const renderRow = useCallback(
+      ({ item, index }) => {
+        const { id } = item;
+        const isSelected = extraData?.[id];
+        const rowStyle = isSelected
+          ? selectedRowStyle
+          : (index % 2 === 0 && alternateRowStyle) || basicRowStyle;
 
-  return (
-    <FlatList
-      data={data}
-      renderItem={renderRow}
-      stickyHeaderIndices={[0]}
-      ListHeaderComponent={renderHeaders}
-      extraData={extraData}
-    />
-  );
-});
+        return (
+          <SimpleRow
+            rowData={item}
+            rowIndex={index}
+            rowKey={id}
+            style={rowStyle}
+            renderCells={renderCells}
+            onPress={selectRow}
+            isSelected={isSelected}
+          />
+        );
+      },
+      [extraData]
+    );
+
+    const renderHeaderCells = useCallback(
+      () =>
+        columns.map(({ title, width, alignText }, index) => (
+          <HeaderCell
+            key={title}
+            title={title}
+            containerStyle={headerCells[alignText]}
+            textStyle={cellText[alignText]}
+            isLastCell={index === columns.length - 1}
+            width={width}
+          />
+        )),
+      [columns]
+    );
+
+    const renderHeaders = useCallback(
+      () => <HeaderRow style={headerRow} columns={columns} renderCells={renderHeaderCells} />,
+      [columns]
+    );
+
+    return (
+      <FlatList
+        ref={ref}
+        data={data}
+        keyExtractor={recordKeyExtractor}
+        getItemLayout={getItemLayout}
+        renderItem={renderRow}
+        stickyHeaderIndices={[0]}
+        ListHeaderComponent={renderHeaders}
+        extraData={extraData}
+      />
+    );
+  })
+);
+
+SimpleTable.defaultProps = {
+  extraData: {},
+};
 
 SimpleTable.propTypes = {
-  data: PropTypes.array.isRequired,
+  data: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
```

Comment:
The page comment still refers to `data` as an array only

Reply:
And `FlatList` says that it takes a `ReadOnlyArray` for the `data` property - presumably an object is read as an array of properties and just works?

Reply:
Tried to clarify here: f89f49ad74a85c1d4ef2db760270bf7042947c40

A "realm array" or `Realm.results` is an object `{ 0, 1, 2, .. n }` so is an array that works.. can we get rid of realm yet

Reply:
oh. sorry I asked